src/Generators/randomgenerators.jl

```diff
@@ -0,0 +1,407 @@
+abstract type RandomGenerator <: GraphGenerator end
+
+"""
+    struct ErdosRenyi <: RandomGenerator
+
+A struct representing a generator for an [Erdős–Rényi](http://en.wikipedia.org/wiki/Erdős–Rényi_model)
+random graph with `n` vertices. Edges are added between pairs of vertices with
+probability `p`.
+
+### Required Fields
+- `n::Integer`: the number of vertices in the graph
+- `p::Real`: the probability of edge creation (`0 < p ≤ 1`)
+
+### Optional Arguments
+- `rng::AbstractRNG`: set the random number generator (default: `Random.GLOBAL_RNG`)
+"""
+struct ErdosRenyi{T<:Integer, U<:Real, R<:AbstractRNG} <: RandomGenerator
+    n::T
+    p::U
+    rng::R
+end
+ErdosRenyi(n, p; rng=GLOBAL_RNG) = ErdosRenyi(n, p, rng)
+
+"""
+    struct ApproxErdosRenyi <: RandomGenerator
+
+Create a generator approximating an [Erdős–Rényi](http://en.wikipedia.org/wiki/Erdős–Rényi_model) random
+graph with `n` vertices and `ne` edges.
+
+### Required Fields
+- `n::Integer`: the number of vertices in the graph
+- `ne::Integer`: the number of edges in the graph
+
+### Optional Arguments
+- `rng::AbstractRNG`: set the random number generator (default: `Random.GLOBAL_RNG`)
+"""
+struct ApproxErdosRenyi{T<:Integer, U<:Integer, R<:AbstractRNG} <: RandomGenerator
+    n::T
+    ne::U
+    rng::R
+end
+ApproxErdosRenyi(n, ne; rng=GLOBAL_RNG) = ApproxErdosRenyi(n, ne, rng)
```

Comment:
Maybe we could call this one `ErdosRenyiGnm`. Networkx does something similar.

Reply:
Since they're actually both valid E-R implementations, I think we should just merge them into one with different parameters. It's easy enough to do, I think.

Reply:
OK, The old `ErdosRenyi` is now called `Binomial` (after NetworkX), and this one with `ne` is `ErdosRenyi`.